django/thunderstore/repository/models/namespace.py

```diff
@@ -0,0 +1,42 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models import Manager
+
+from thunderstore.repository.validators import PackageReferenceComponentValidator
+
+
+class Namespace(models.Model):
+    objects: "Manager[Namespace]"
+
+    name = models.CharField(
+        max_length=64,
+        unique=True,
+        validators=[PackageReferenceComponentValidator("Namespace name")],
+    )
+    team = models.ForeignKey(
+        "repository.Team",
+        on_delete=models.CASCADE,
```

Comment:
Definitely should not cascade, namespaces are immutable & read-only

Reply:
Whatthehell, I thought I put it as PROTECT. Welp, good that you noticed this :D

Reply:
I actually changed this to SET_NULL and enabled null=True and blank=True on the Namespace model. Now we can delete the team and the namespace can live on.